docs/changes/table/13129.feature.rst

```diff
@@ -0,0 +1 @@
+Added support for reading and writing files where the file path is of the form ``~/data/file.csv``, where ``~`` indicates the home directory.
```

Comment:
You need to make it clear that this applies the Unified I/O layer and not (for instance) to direct I/O within the `fits` subpackage. Note that NDData and Cosmology also support I/O through this interface and should likewise now allow `~`.

Also, this should make note that `~user/data/file.csv` also works.

Finally, it makes it a little easier to read/review if the text here is wrapped at 80  columns.